### DISCUSSION TITLE
Different color for component icon

### DISCUSSION THREAD

**Discussion by fitrh** [nvim-lualine/lualine.nvim#558] [Ideas]
![image](https://user-images.githubusercontent.com/55179750/150661489-cf45e868-fecd-4e1d-8e80-105a1ee3473d.png)

This idea comes from [akinsho/flutter-tools.nvim screenshot](https://github.com/akinsho/flutter-tools.nvim).

So we can have something like this

```lua
-- just like padding, if string, then use the component highlight group   
icon = { "", color = { fg = "#000", bg = "#111" } }
--                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
--                      or highlight-group name
```

or this

```lua
icon = ""
icon_color = { fg = "#000", bg = "#111" }
```

Not a functional feature, just an eye-candy.

**Comment by shadmansaleh** [comment]
I'll consider this.

For now you can use separate component for the icon to get the look.

```lua
lualine_c = {
   {'', type='stl', color={fg='#f1502f'}, separator='', padding={left=1}},
   {'branch', color = { fg = '#30a0f0' }, icons_enabled=false },
}
```

**Comment by shadmansaleh** [comment]
@fitrh  added in 32c85cd214c387fd4883ccb18257e0204184fb85

Now you can do things like
```lua
{'branch', icon = {'', color={fg='green'}}}
```

**Reply by fitrh** [reply]
Thanks, I'll give it a try

**Reply by fitrh** [reply]
I have tried this feature, it seems that icon color requires `bg` to be set, unlike component color which will use default `StatusLine` background if `bg` is not set

```lua
{
    "branch",
    icon = { "", color = { fg = palette.white.secondary } },
    color = { fg = palette.magenta.primary, gui = "BOLD" },
}
```

![20220308_000825_area](https://user-images.githubusercontent.com/55179750/157074329-7fca873e-2a52-4119-939b-4eb8bfab3a7e.jpg)

**Reply by shadmansaleh** [reply]
That's strange both colors are created and used in same manner. Maybe you're hitting some edge case I'll look in to it later . Can you provide your lualine theme + values of highlight groups `Normal`, `Statusline`, `StatuslineNC`

**Reply by fitrh** [reply]
[The theme](https://github.com/fitrh/init.nvim/blob/main/lua/plugin/themes/tokyonight/lualine.lua) is a slightly modified of tokyonight
```
:hi Normal
Normal         xxx guifg=#c0caf5 guibg=#1a1b26
```

```
:hi StatusLine
StatusLine     xxx cterm=bold,reverse guifg=#a9b1d6 guibg=#16161e
```

```
:hi StatusLineNC
StatusLineNC   xxx cterm=reverse guifg=#3b4261 guibg=#16161e
```
